### ISSUE TITLE
[WinUI] TemplatedItemsList and underlying ListView UI elements are disconnected, creates phantom views.

### ISSUE THREAD

**Issue by drasticactions** [dotnet/maui#18401]
### Description

TL;DR Every time you add a new item to a WinUI ListView that's bound to an ObservableCollection, two Cells are created and only one is outputted to the screen. In order to fix https://github.com/dotnet/maui/issues/13900, we need to address this above.

- When a MAUI WinUI ListView gets created, the attached virtual ItemSource is taken, transformed into an ObservableCollection if it's not already, and set as a CollectionViewSource for the underlying WinUI ListView, https://github.com/dotnet/maui/blob/3c0ff6735c6b259a17cdef31f7ebce8dea1e0a30/src/Compatibility/Core/src/Windows/ListViewRenderer.cs#L127-L164
- If you already have items in your item source, the VisualRenderers will start to invoke and request elements, which in turn causes WinUI to invoke elements to be created. These will result in the first `Cell` items to be created and put into the WinUI ListView. Note that these cells are not directly referenced or run through TemplatedItemsList, so according to it, these Cells are `null`.

<img width="518" alt="initial" src="https://github.com/dotnet/maui/assets/898335/9eb1d2be-7c89-4af5-8cd2-3c1ce466de33">

- If you add another item to the ItemSource (and the ItemSource is observable),  a `OnProxyCollectionChanged` handler event is fired, https://github.com/dotnet/maui/blob/3c0ff6735c6b259a17cdef31f7ebce8dea1e0a30/src/Controls/src/Core/TemplatedItemsList.cs#L998-L1007
- This creates a new `Cell` based on the underlying template.

<img width="717" alt="newcell" src="https://github.com/dotnet/maui/assets/898335/f94a69a0-94ac-4402-bd11-b3257d7114c9">

- Once that's done, another OnCollectionChanged is fired to the ListViewRenderer. https://github.com/dotnet/maui/blob/3c0ff6735c6b259a17cdef31f7ebce8dea1e0a30/src/Controls/src/Core/TemplatedItemsList.cs#L1104
- While the event contains the new ViewCell as created in the last method, this method doesn't use it. Instead, it calls on the underlying WinUI List to update its layout (which, based on the ItemsSource having the new items, will then generate cells)
- https://github.com/dotnet/maui/blob/3c0ff6735c6b259a17cdef31f7ebce8dea1e0a30/src/Compatibility/Core/src/Windows/ListViewRenderer.cs#L238
- This will then call for _new_ `Cell` objects that are then inserted in the real WinUI list.

<img width="747" alt="RealCell" src="https://github.com/dotnet/maui/assets/898335/3eaef4fd-405a-4507-8abb-61c6f6b1e952">

This, I believe, creates two separate lists of given Cells: One in the native WinUI view, and one in TemplatedItemsList, that are not directly linked, with TemplatedItemsList generating Cells that point to nothing. Beyond potential memory considerations, this causes issues with tooling. New Items inserted into an ItemsSource are processed twice, once in TemplatedItemsList and again by WinUI, creating two distinct Cells. 

https://github.com/dotnet/maui/assets/898335/1a6b3737-c499-416f-9198-16171440c0dd

You can see this manifest with XAML Hot Reload. The initial items in the ListView are not represented in the Live Visual Tree, because they are never registered to VisualDiagnostics.OnChildrenAdded. That gets called in SetupContent, which isn't called for initial items in the list.

https://github.com/dotnet/maui/blob/3c0ff6735c6b259a17cdef31f7ebce8dea1e0a30/src/Controls/src/Core/ListView/ListView.cs#L441-L453

New items do get called through TemplatedItemsList, so they do invoke SetupContent and do get added to LVT. However, these cells are not the ones shown to the user. If you try to make Hot Reload changes to them, while they appear in LVT, they do not appear in the running app. This is because there are two Cells for every given item, and we're editing the one within TemplatedItemsList, which is not the one shown on screen. Those items do exist and XAML Hot Reload is doing the correct thing here.

This happens in all versions of MAUI.

### Steps to Reproduce

Copy the reproduction project into the MAUI Sandbox Project in the repo.

### Link to public reproduction project repository

https://github.com/drasticactions/MauiRepros/tree/main/MauiListViewTest

### Version with bug

7.0.96

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

**Comment by drasticactions** [comment]
Updated the tags to include hot-reload-xaml. To make it clear, I believe that to address ListView for WinUI not being able to be hot reloaded (https://github.com/dotnet/maui/issues/13900) we need to fix this issue. This issue is blocking ListView Hot Reload from fully working on MAUI with WinUI.

**Comment by kevinxufei** [comment]
Verified this issue with Visual Studio 17.10.0 Preview 3 (8.0.3/8.0.20). Can not repro this issue with sample project.
